secret/scanner.go

```diff
@@ -72,6 +141,34 @@ func (s Scanner) Scan(args ScanArgs) types.Secret {
 	}
 }
 
+func removeAllowedSecrets(allRuleResults []RuleResult, allowedLocations [][]int) []RuleResult {
+	if len(allowedLocations) > 0 {
+		results := make([]RuleResult, 0)
+		for _, result := range results {
```

Comment:
You may want to iterate allRuleResults here. Iterating `results` doesn't make sense.